tests/bson/bson-iterator-clone-002.phpt

```diff
@@ -0,0 +1,22 @@
+--TEST--
+MongoDB\BSON\Iterator is rewound when cloning
```

Comment:
Is this desirable, or do you think we should inherit the original object's offset?

Reply:
A quick check shows the same behaviour applies for `ArrayIterator`. I honestly haven't given this too much thought, but I figured that if a user wants to clone the iterator, they may be interested in iterating it from the start.